src/server.py

```diff
@@ -464,37 +420,97 @@ def run_seta_details_query():
+    alljobs = JOBSDATA.jobtype_query()
+
+    # For the case of TaskCluster request, we don't care which priority the user request.
+    # We return jobs depend on the strategy that we return high value jobs as default and
+    # return all jobs for every 5 push or 90 minutes for that branch.
     if request.headers.get('User-Agent', '') == 'TaskCluster':
 
+        # we query all jobs rather than jobs filter by the requested priority in here,
+        # Because we need to set the job returning strategy depend on different job priority.
+        query = session.query(JobPriorities.platform,
+                              JobPriorities.buildtype,
+                              JobPriorities.testtype,
+                              JobPriorities.priority,
+                              JobPriorities.timeout
+                              ).all()
+
         # We should return full job list as a fallback, if it's a request from
         # taskcluster and without head_rev or pushlog_id in there
-        if head_rev or pushlog_id:
-            priority = valve(head_rev, pushlog_id, branch, priority)
+        branch_info = session.query(TaskRequests.counter, TaskRequests.datetime).filter(
+            TaskRequests.branch == branch).all()
+
+        time_of_now = datetime.datetime.now()
+        time_of_request = time.strftime("%Y-%m-%d %H:%M:%S")
+
+        # If we got nothing related with that branch, we should create it.
+        if len(branch_info) == 0:
+            # time_of_lastreset is not a good name anyway :(
+            # And we treat all branches' reset_delta is 90 seconds, we should find a
+            # better delta for them in the further.
+            branch_data = TaskRequests(str(branch), 1, str(time_of_request), 5400)
+            try:
+                session.add(branch_data)
+                session.commit()
+            except Exception:
+                session.rollback()
+
+            finally:
+                session.close()
+            counter, string_time, reset_delta = 1, time_of_request, 5400
+
+        # We should update it if that branch had already been stored.
         else:
-            priority = 0
+            counter, string_time, reset_delta = branch_info[0]
+            conn = engine.connect()
+            statement = update(TaskRequests).where(
+                TaskRequests.branch == branch).value(
+                counter == counter + 1)
+            conn.execute(statement)
+
+        last_update_time = datetime.datetime.strptime(string_time, "%Y-%m-%d %H:%M:%S")
+        delta = (time_of_now - last_update_time).total_seconds()
+
+        # we should update the time recorder if the elapse time had
+        # reach the time limit of that branch.
+        if delta >= reset_delta:
+            conn = engine.connect()
+            statement = update(TaskRequests).where(
+                TaskRequests.branch == branch).value(
+                datetime == datetime)
+            conn.execute(statement)
```

Comment:
I would like to update the database only once here- to reduce operations.  We now have:
query for jobs
query for branch data
insert branch data (rare)
increment counter
update timestamp

I would like to increment counter and update timestamp at the same time- we should update the timestamp for all calls, the reset would force us to run all jobs instead of use the specific priority.